contracts/extensions/FundingQueue.sol

```diff
@@ -0,0 +1,349 @@
+/*
+  This file is part of The Colony Network.
+
+  The Colony Network is free software: you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation, either version 3 of the License, or
+  (at your option) any later version.
+
+  The Colony Network is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with The Colony Network. If not, see <http://www.gnu.org/licenses/>.
+*/
+
+pragma solidity 0.5.8;
+pragma experimental ABIEncoderV2;
+
+import "./../../lib/dappsys/math.sol";
+import "./../colony/IColony.sol";
+import "./../colonyNetwork/IColonyNetwork.sol";
+import "./../common/ERC20Extended.sol";
+import "./../patriciaTree/PatriciaTreeProofs.sol";
+import "./../tokenLocking/ITokenLocking.sol";
+
+
+contract FundingQueue is DSMath, PatriciaTreeProofs {
+
+  // Events
+  event ProposalCreated(uint256 id, uint256 indexed fromPot, uint256 indexed toPot, address indexed token, uint256 amount);
+  event ProposalStaked(uint256 indexed id, uint256 domainTotalRep);
+  event ProposalBacked(uint256 indexed id, uint256 indexed newPrevId, uint256 userRep);
+  event ProposalPinged(uint256 indexed id, uint256 amount);
+  event ProposalCompleted(uint256 indexed id);
+  event ProposalCancelled(uint256 indexed id);
+
+  // Constants
+  uint256 constant HEAD = 0;
+  uint256 constant UINT256_MAX = (2 ** 256) - 1;
+  uint256 constant STAKE_PCT = WAD / 1000; // 0.1%
```

Comment:
Contender for worst-named variable I've seen this year, I have to say. It's called 'Stake Percent', the comment suggests it represents 0.1%, and it holds a value of `1000000000000000` :P